showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITHttpAnnotation.java

```diff
@@ -0,0 +1,103 @@
+
+  @Before
+  public void createClient() throws IOException, GeneralSecurityException {
+    ComplianceSuite.Builder builder = ComplianceSuite.newBuilder();
+    JsonFormat.parser()
+        .merge(
+            new InputStreamReader(
+                ITHttpAnnotation.class
+                    .getClassLoader()
+                    .getResourceAsStream("compliance_suite.json")),
+            builder);
+    complianceSuite = builder.build();
+
+    ComplianceSettings httpjsonComplianceSettings =
+        ComplianceSettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                ComplianceSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    complianceClient = ComplianceClient.create(httpjsonComplianceSettings);
+
+    // Mapping of Compliance Suite file RPC Names to ComplianceClient methods
+    complianceValidRpcMap =
+        ImmutableMap.of(
+            "Compliance.RepeatDataBody",
+            complianceClient::repeatDataBody,
+            "Compliance.RepeatDataBodyInfo",
+            complianceClient::repeatDataBodyInfo,
+            "Compliance.RepeatDataQuery",
+            complianceClient::repeatDataQuery,
+            "Compliance.RepeatDataSimplePath",
+            complianceClient::repeatDataSimplePath,
+            "Compliance.RepeatDataBodyPut",
+            complianceClient::repeatDataBodyPut,
+            "Compliance.RepeatDataBodyPatch",
+            complianceClient::repeatDataBodyPatch,
+            "Compliance.RepeatDataPathResource",
+            complianceClient::repeatDataPathResource);
+  }
+
+  @After
+  public void destroyClient() {
+    complianceClient.close();
+  }
+
+  /*
+  This test loops through all the testing groups (which test different aspects of the http annotation)
+  and calls the listed RPC Names (which are the different HttpVerbs) with the given request. The response's
+  info should be the same as the request's. This tests that the request is set correctly into the corresponding
+  URL/Body/Query.
+   */
+  @Test
+  public void testHttpAnnotations() {
+    for (ComplianceGroup compliancegroup : complianceSuite.getGroupList()) {
+      List<String> validRpcList =
+          compliancegroup.getRpcsList().stream()
+              .filter(complianceValidRpcMap::containsKey)
+              .collect(Collectors.toList());
+      for (String rpcName : validRpcList) {
+        for (RepeatRequest repeatRequest : compliancegroup.getRequestsList()) {
+          ComplianceData expectedData = repeatRequest.getInfo();
+          RepeatResponse response = complianceValidRpcMap.get(rpcName).apply(repeatRequest);
```

Comment:
I like how you've structured this with the map.

Minor suggestion: I got caught on this line having to figure out what was going on. Consider the slightly more explicit:

```java
for (String rpcName : validRpcList) {
    Function<Request, Response> rpc = complianceValidRpcMap.get(rpcName);
    for (Request request : getRequestsList()) {
        Response response = rpc.apply(request);
        ...
    }

Reply:
Sure, will do!